### DISCUSSION TITLE
Is it possible to extend or inherit templates?

### DISCUSSION THREAD

**Discussion by deviantintegral** [lima-vm/lima#2439] [Q&A] [answered]
I'm comparing how lima and colima manage their configuration. One of the nice things I've found about colima's templates is that they aren't tied to the underlying OS image. This allows the OS image to be updated (for new instances) without having to change the configuration containing default CPU settings, provision scripts, and the like.

The closest I see with lima is using `limactl start --set` to pass in yq directives, but that is going to be painful when inlining provision scripts.

Any suggestions?

**Comment by afbjorklund** [comment, marked as answer]
Doing some experiments with separating them, e.g. `images: [default]`:

* https://github.com/lima-vm/lima/issues/2418

But it is mostly about abstracting/simplifying the syntax, for the YAML.

As you can see, the current templates are using copy/paste inheritance.
